Sources/Connection.swift

```diff
@@ -0,0 +1,20 @@
+//
+//  Connection.swift
+//  C7
+//
+//  Created by Scott Byrns on 4/1/16.
+//
+//
+
+public protocol Connection {
+    
+    associatedtype StatusType
+    
+    func open() throws
+    func close() -> Bool
+    
+    var connectionInfo: ConnectionInfo { get }
+    var status: StatusType { get }
+    var mostRecentError: ErrorProtocol? { get }
```

Comment:
`mostRecentError` doesn't seem too swifty. in which scenario wouldn't I be able to extract this error from the action itself, that I need to poll this property?